### PR TITLE
Espresso for logisim

### PR DESCRIPTION
**Pull request by BFH-ktt1**
DEPENDS_ON #1511

### CODE REVIEW

File: src/main/java/com/cburch/logisim/analyze/data/EspressoCubeStruct.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+import com.cburch.logisim.analyze.model.AnalyzerModel;
+
+public class EspressoCubeStruct {
+
+  // TODO: This code has not been optimized for performance, it just tries to implement espresso for logisim in java.
+  //       For sure a lot can be done here.
+  // Important: The original code is based on "unsigned int" for storage, as java does not know the unsigned version I
+  //            tried to adapt to int's, however I might have left some "bugs"
+  // Note: The (variable/function) names have been kept as close as possible to the original onces keeping in mind logisim
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/analyze/data/EspressoCubeStruct.java:37]: "onces" -> "ones"

File: src/main/java/com/cburch/logisim/analyze/model/VariableList.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public boolean containsDuplicate(VariableList data, Var oldVar, String name) {
   }
 
   public void add(Var variable) {
-    if (data.size() + variable.width >= maxSize) {
+    if (data.size() + variable.width > maxSize) {
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/analyze/model/VariableList.java:61]: This looks like a bug, which could already be fixed independent of this PR.

**Review by maehne — COMMENTED** [review]
Some of the new classes lack documentation. If possible, use Javadoc-style comments. Document at least the purpose of each class if you don't have time documenting each member function. In `EspressoCubeStruct.java`, some of your comments are hard to read, as you don't follow common conventions like starting new sentences with a capital letter and some punctuation marks are missing.

**Comment by BFH-ktt1** [comment]
No time to dive further in this, closing

**Pull request closed without merging**